lib/GHCup/GHC.hs

```diff
@@ -868,20 +868,25 @@ compileGHC targetGhc crossTarget vps bstrap hghc jobs mbuildConfig patches aargs
         tmpDownload <- lift withGHCupTmpDir
         tmpUnpack <- lift mkGhcupTmpDir
         tar <- liftE $ download uri Nothing Nothing Nothing (fromGHCupPath tmpDownload) Nothing False
-        (bf, tver) <- liftE $ cleanUpOnError @'[UnknownArchive, ArchiveResult, ProcessError] tmpUnpack $ do
+        (workdir, tver) <- liftE $ cleanUpOnError @'[UnknownArchive, ArchiveResult, ProcessError, PatchFailed, DownloadFailed, DigestError, ContentLengthError, GPGError, NotFoundInPATH] tmpUnpack $ do
           liftE $ unpackToDir (fromGHCupPath tmpUnpack) tar
-          let regex = [s|^(.*/)*boot$|] :: B.ByteString
-          [bootFile] <- liftIO $ findFilesDeep
+
+          let regex = [s|^(.*/)*compiler/ghc.cabal.in$|] :: B.ByteString
```

Comment:
can you explain this?

Reply:
This change is to fix the `workdir` retrieval. The bootstrapped src does not contain `boot` file.

finding the directory name using `ghc.cabal.in` seems to be better.

Reply:
I'm not sure this file will always exist in that place. If there is no `boot` file, then we can assume there is a `configure` file. And the topmost such configure file is what we need. We should depend on the files we actually use.

From what I understand `findFilesDeep` is breadth first, so we can just take the first find (possibly using the conduit functions to cut off early).

Reply:
I reverted back to `boot` file, the file gets renamed now in hadrian when making release builds.

```
      -- And move ./boot so we can't accidentally call it in CI                                                           
      moveFile (dest -/- "boot") (dest -/- "boot.source")       
```